src/graphql/node/input.rs

```diff
@@ -123,8 +132,31 @@ pub(super) fn create_draft_update(
         None
     };
 
-    let agents: Vec<review_database::Agent> = if let Some(agents) = new.agents {
-        agents.into_iter().map(Into::into).collect()
+    let old_config_map: HashMap<String, Option<String>> = old
+        .agents
+        .iter()
+        .map(|agent| (agent.key.clone(), agent.config.clone()))
+        .collect();
+
+    let agents: Vec<review_database::Agent> = if let Some(new_agents) = new.agents {
+        new_agents
+            .into_iter()
+            .map(|agent_draft| {
+                let config = old_config_map
+                    .get(&agent_draft.key)
+                    .cloned()
+                    .and_then(|config| config.and_then(|c| c.try_into().ok()));
```

Comment:
This is good enough. However, I would suggest using `clone` as late as possible in the code, as it helps convey why `clone` is inevitable.